docker-app/qfieldcloud/core/views/files_views.py

```diff
@@ -135,25 +139,50 @@ def get(self, request, projectid, filename):
 
     # TODO refactor this function by moving the actual upload and Project model updates to library function outside the view
     def post(self, request, projectid, filename, format=None):
-        project = Project.objects.get(id=projectid)
+        """Handle POST requests."""
+
+        if len(request.FILES.getlist("file")) > 1:
+            raise exceptions.MultipleContentsError()
+
+        # QF-2540
+        # Getting traceback in case the traceback provided by Sentry is too short
+        # Add post-serialization keys for diff-ing with pre-serialization keys
+        buffer = io.StringIO()
+        print_stack(limit=50, file=buffer)
+        request_attributes = {
+            "data": str(copy.copy(self.request.data).keys()),
```

Comment:
Is there a need for a copy, I think the `str(...)` conversion will already copy anyway

Reply:
Under the hood request.data is apparently not idempotent -- calling it once entirely consumes it, calling it twice yield None. I am copying to avoid consuming it.